### PR TITLE
New configuration tool

### PR DESCRIPTION
**Pull request by jasonleenaylor**
Changes related to creating a new configuration tool for the windows installers that doesn't require us to have an old version of IE installed with ActiveX controls enabled.

[Review on Reviewable](https://reviewable.io/reviews/sillsdev/masterinstaller/1)

### CODE REVIEW

File: Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		public class SetupOptions
+		{
+			[XmlElement(ElementName = "CppFilePath")]
+			public string CppPath { get; set; }
+
+			[XmlElement(ElementName = "ProductsPath")]
+			public string ProductsPath { get; set; }
+
+			[XmlElement(ElementName = "CDsPath")]
+			public string CdPath { get; set; }
```
Comment (papeh) [inline Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs:50]: `Cd` vs `CDs` (fine as long as it's intentional)

**Comment by papeh** [comment]
Reviewed 33 of 33 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

<sup>**[Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs, line 106 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28h0hcA8TbgNAAlC9J)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs#L106)):</sup>
I may have made both ...NeedsShiftCtrl



---

<sup>**[Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs, line 119 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28ievXO49jFsmMZvCn)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs#L119)):</sup>
On line 35, the XmlElementName is `TermsOfUse`, but the Property name is`Terms`; here, the opposite



---

<sup>**[Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs, line 13 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28jk-sreLj6qf2XlqG)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs#L13)):</sup>
SIL International?



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

**Comment by jasonleenaylor** [comment]
<sup>**[Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs, line 13 [r1]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K28jk-sreLj6qf2XlqG)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/32f24fec1f45f03313f1d9a76eb5eec56477f1fd/Code and Projects/MasterInstallerConfigurator/Properties/AssemblyInfo.cs#L13)):</sup>
Yes, I need to do this in a further commit.



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

**Comment by papeh** [comment]
Reviewed 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs, line 194 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2ClZ-Sh-PHoCT8rUEk)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfigurator/ConfigurationModel.cs#L194)):</sup>
Theoretically, a client could set ValueString to a non-int; however, I'm not sure how much we need to worry about this problem in this project.



---

<sup>**[Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs, line 99 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cr1WiSWhawFiQONc0)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs#L99)):</sup>
I assume this will be filled in later?



---

<sup>**[Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs, line 86 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cqb41CofRFo0fgs2N)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs#L86)):</sup>
I don't see where these true and false values are used



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

**Comment by jasonleenaylor** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs, line 99 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cr1WiSWhawFiQONc0)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfigurator/JavaScriptConverter.cs#L99)):</sup>
Yes.



---

<sup>**[Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs, line 86 [r3]](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1#-K2Cqb41CofRFo0fgs2N)** ([raw file](https://github.com/sillsdev/masterinstaller/blob/8627b37478b2aa1c585beba423726f7067a7e16a/Code and Projects/MasterInstallerConfiguratorTests/MasterInstallerConfiguratorTests/JavaScriptConverterTests.cs#L86)):</sup>
In the FlavorProducts line 75



---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sillsdev/masterinstaller/1)

**Pull request merged**